pyccl/tests/test_profiles.py

```diff
@@ -0,0 +1,231 @@
+import numpy as np
+import pytest
+import pyccl as ccl
+
+
+COSMO = ccl.Cosmology(
+    Omega_c=0.27, Omega_b=0.045, h=0.67, sigma8=0.8, n_s=0.96,
+    transfer_function='bbks', matter_power_spectrum='linear')
+M200 = ccl.halos.MassDef200c()
+
+
+def one_f(cosmo, M, a=None, mdef=None):
+    if np.ndim(M) == 0:
+        return 1
+    else:
+        return np.ones(M.size)
+
+
+def smoke_assert_prof_real(profile):
+    sizes = [(0, 0),
+             (2, 0),
+             (0, 2),
+             (2, 3),
+             (1, 3),
+             (3, 1)]
+    shapes = [(),
+              (2,),
+              (2,),
+              (2, 3),
+              (1, 3),
+              (3, 1)]
+    for (sm, sr), sh in zip(sizes, shapes):
+        if sr == 0:
+            r = 0.5
+        else:
+            r = np.linspace(0., 1., sr)
+        if sm == 0:
+            m = 1E12
+        else:
+            m = np.geomspace(1E10, 1E14, sm)
+        p = profile._real(COSMO, r, m, 1., M200)
+        assert np.shape(p) == sh
+
+
+def test_defaults():
+    p = ccl.halos.HaloProfile()
+    with pytest.raises(NotImplementedError):
+        p.real(None, None, None, None)
+    with pytest.raises(NotImplementedError):
+        p.fourier(None, None, None, None)
+
+
+@pytest.mark.parametrize('prof_class',
+                         [ccl.halos.HaloProfileNFW,
+                          ccl.halos.HaloProfileHernquist,
+                          ccl.halos.HaloProfileEinasto])
+def test_empirical_smoke(prof_class):
+    c = ccl.halos.ConcentrationDuffy08(M200)
+    with pytest.raises(TypeError):
+        p = prof_class(None)
+
+    if prof_class == ccl.halos.HaloProfileNFW:
+        with pytest.raises(ValueError):
+            p = prof_class(c,
+                           projected_analytic=True,
+                           truncated=True)
+
+    p = prof_class(c)
+    smoke_assert_prof_real(p)
+
+
+@pytest.mark.parametrize('prof_class',
+                         [ccl.halos.HaloProfileGaussian,
+                          ccl.halos.HaloProfilePowerLaw])
+def test_simple_smoke(prof_class):
+    def r_s(cosmo, M, a, mdef):
+        return mdef.get_radius(cosmo, M, a)
+
+    p = prof_class(r_s, one_f)
+    smoke_assert_prof_real(p)
+
+
+def test_gaussian_accuracy():
+    def fk(k):
+        return np.pi**1.5 * np.exp(-k**2 / 4)
+
+    p = ccl.halos.HaloProfileGaussian(one_f, one_f)
+
+    k_arr = np.logspace(-3, 2, 1024)
+    fk_arr = p.fourier(COSMO, k_arr, 1., 1.)
+    fk_arr_pred = fk(k_arr)
+    res = np.fabs(fk_arr - fk_arr_pred)
+    assert np.all(res < 5E-3)
+
+
+@pytest.mark.parametrize('alpha', [-1.2, -2., -2.8])
+def test_projected_plaw(alpha):
```

Comment:
probably should be named `test_projected_plaw_accuracy`

Reply:
Done